File: app/src/main/java/com/raiadnan/quranreader/hadith/settings/SettingsActivity.java
package com.raiadnan.quranreader.hadith.settings;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.preference.Preference;
import androidx.preference.PreferenceFragmentCompat;

import com.raiadnan.quranreader.R;

public class SettingsActivity extends AppCompatActivity {
    DrawerLayout drawerLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);
        if (savedInstanceState == null) {
            getSupportFragmentManager()
                    .beginTransaction()
                    .replace(R.id.hadith_host_settings, new SettingsFragment())
                    .commit();
        }
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
        }

        ImageView actionBarSymbol = this.findViewById(R.id.actionbar_left);
        actionBarSymbol.setImageResource(R.drawable.symbol_back);

        actionBarSymbol.setOnClickListener(view -> this.onBackPressed());

        TextView actionBarTitle = this.findViewById(R.id.actionbar_title);
        actionBarTitle.setText(getResources().getString(R.string.title_settings));

        ImageView actionbarBookmark = this.findViewById(R.id.actionbar_bookmark);
        actionbarBookmark.setVisibility(View.GONE);
        ImageView actionBarSearch = this.findViewById(R.id.actionbar_search);
        actionBarSearch.setVisibility(View.GONE);
        ImageView actionBarSettings = this.findViewById(R.id.actionbar_settings);
        actionBarSettings.setVisibility(View.GONE);



    }

    public static class SettingsFragment extends PreferenceFragmentCompat {
        @Override
        public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
            setPreferencesFromResource(R.xml.root_preferences, rootKey);
        }
    }

    @Override
    public void onBackPressed() {
super.onBackPressed();
    }
}

File: app/src/main/java/com/raiadnan/quranreader/ui/Main/View/SettingsActivity.kt
package com.raiadnan.quranreader.ui.Main.View

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import androidx.preference.SwitchPreference
import com.raiadnan.quranreader.R
import com.raiadnan.quranreader.Utils.SharePref

class SettingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.settings_activity)
        if (savedInstanceState == null) {
            supportFragmentManager
                .beginTransaction()
                .replace(R.id.settings, SettingsFragment())
                .commit()
        }
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
    }

    class SettingsFragment : PreferenceFragmentCompat() {

        var sharePref: SharePref? = null

        companion object {
            const val WORKER_TAG = "notification_worker"
        }

        override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
            setPreferencesFromResource(R.xml.root_preferences, rootKey)
            sharePref = SharePref(requireContext())
            val switchPreference =
                findPreference<SwitchPreference>(requireContext().getString(R.string.pref_notification_key))
            switchPreference!!.isChecked = sharePref!!.loadSwitchState()
        }

        override fun onPreferenceTreeClick(preference: Preference): Boolean {
            val notificationKey = getString(R.string.pref_notification_key)
            /*val listKey = getString(R.string.pref_sort_key)*/
            if (preference.key == notificationKey) {
                val on =
                    (preference as SwitchPreference).isChecked
                sharePref!!.saveSwitchState(on)
            }
            return super.onPreferenceTreeClick(preference)
        }
    }
}